modules/loki/main.tf

```
@@ -1,11 +1,162 @@
 locals {
-  chart_name    = "loki-stack"
-  chart_version = var.chart_version
-  release_name  = "loki"
-  namespace     = var.namespace
-  repository    = "https://grafana.github.io/loki/charts"
+  chart_name     = "loki-stack"
+  chart_version  = var.chart_version
+  release_name   = "loki"
+  namespace      = var.namespace
+  repository     = "https://grafana.github.io/loki/charts"
+  provider_url   = replace(var.oidc_provider_issuer_url, "https://", "")
+  bucket_name    = "${var.cluster_name}-loki-${data.aws_caller_identity.loki.account_id}"
+  dynamodb_table = "${var.cluster_name}-loki"
+
+  loki_storage_s3 = {
+    loki = {
+      config = {
+        schema_config = {
+          configs : [
+            {
+              from         = "2020-05-15"
+              store        = "aws"
+              object_store = "s3"
+              schema       = "v11"
+              index = {
+                prefix = local.dynamodb_table
+              }
+            }
+          ]
+        }
+        # Read more here: https://github.com/grafana/loki/tree/master/docs/configuration#storage_config
+        storage_config = {
+          aws = {
+            s3 = "s3://${data.aws_region.loki.name}/${local.bucket_name}"
+            dynamodb = {
+              dynamodb_url = "dynamodb://${data.aws_region.loki.name}"
+              metrics = {
+                url: "http://prometheus-server.kube-system.svc.cluster.local:9090"
+              }
+            }
+          }
+        }
+        table_manager = {
+          retention_deletes_enabled = true
+          retention_period = "720h"
+          index_tables_provisioning = {
+            provisioned_write_throughput = 30
+            provisioned_read_throughput = 3
+            write_scale = {
+              enabled = true
+              min_capacity = 1
+              role_arn = module.iam.this_iam_role_arn
+            }
+            read_scale = {
+              enabled = true
+              min_capacity = 1
+              role_arn = module.iam.this_iam_role_arn
+            }
+          }
+        }
+      }
+    }
+  }
 }
 
+data aws_caller_identity "loki" {}
+data aws_region "loki" {}
+
+module "iam" {
+  source = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
```

Purely a cosmetic chore - Double `//`

```suggestion
  source = "terraform-aws-modules/iam/aws/modules/iam-assumable-role-with-oidc"
```

---

The `//` is actually necessary when addressing modules within a module.